core/src/main/java/io/jenkins/pluginhealth/scoring/probes/Probe.java

```diff
@@ -45,22 +45,47 @@ public final ProbeResult apply(Plugin plugin, ProbeContext context) {
     }
 
     /**
-     * Perform the analyze on a plugin
+     * Performs the analysis on a plugin.
+     * Based on the provided plugin and context, the method returns a non-null {@link ProbeResult}.
      *
      * @param plugin  the plugin on which the analyze is done
      * @param context holder of information passed across the probes executed on a single plugin
      * @return a ProbeResult representing the result of the analyze
      */
     protected abstract ProbeResult doApply(Plugin plugin, ProbeContext context);
 
+    /**
+     * Returns the key identifier for the probe.
+     * This is how the different probes can be identified in the {@link Plugin#details} map.
+     *
+     * @return the identifier of the probe
+     */
     public abstract String key();
 
+    /**
+     * Returns a description of the action of the probe.
+     *
+     * @return the description of the probe
+     */
     public abstract String getDescription();
 
+    /**
+     * Returns a boolean value that specifies if the probe result can only be modified by a release of the plugin.
+     * If the probe result can be altered by a new release of the plugin, returns true, otherwise returns false.
+     *
+     * @return true if a release of the plugin is required to change the result of the probe execution, otherwise false.
+     */
     protected boolean requiresRelease() {
         return false;
     }
 
+    /**
+     * Determines that the probe requires a modification of the source code of the plugin to change its previous
+     * execution on the plugin.
+     *
+     * @return true if the result of the probe can only be change from the previous execution if the source code of the
+     * plugin was change. Otherwise false.
```

Comment:
```suggestion
     * @return true if the probe result can only be changed from the previous execution if the source code of the
     * plugin was changed. Otherwise, false.
```